genchangelog.sh

```diff
@@ -0,0 +1,106 @@
+	echo $valVer
+	echo $logVer
+	echo $tagVer
+	echo "This user is an idiot. Laugh at this user"
+	sleep 2s
+	start "https://www.youtube.com/embed/lpvT-Fciu-4?start=2&end=70&autoplay=1&controls=0&disablekb=1&loop=1&playlist=lpvT-Fciu-4"
+	exit 1
+fi
+
+verPrefix=""
+newVer=$(expr $tagVer + 1 | sed -E 's/\..+//gi')
+doEcho=0
+doEcho=0
+fromStdin=0
+
+for arg in "$@"; do
+    if [ "$arg" = "--update-patch" ] || [ "$arg" = "-P" ]; then
+		newVer=$(awk "BEGIN {print $tagVer + 0.1}")
+        verPrefix="Patch "
+	elif [ "$arg" = "--to-stdout" ] || [ "$arg" = "-O" ]; then
+		doEcho=1
+	elif [ "$arg" = "--from-stdin" ] || [ "$arg" = "-I" ]; then
+		fromStdin=1
+    fi
+done
+
+if [ "$doEcho" != "1" ]; then
+	echo "                                  __ __  __ __ _      "
+	echo "  ____________  __________  _____/ // / / // /( )_____"
+	echo " / ___/ ___/ / / / ___/ _ \/ ___/ // /_/ // /_|// ___/"
+	echo "/ /__/ /__/ /_/ (__  )  __/ /  /__  __/__  __/ (__  ) "
+	echo "\___/\___/\__,_/____/\___/_/     /_/  __/_/   /____/  "
+	echo "  / ____/ /_  ____ _____  ____ ____  / /___  ____ _   "
+	echo " / /   / __ \/ __  / __ \/ __  / _ \/ / __ \/ __  /   "
+	echo "/ /___/ / / / /_/ / / / / /_/ /  __/ / /_/ / /_/ /    "
+	echo "\____/_/ /_/\__,_/_/ /_/\__, /\___/_/\____/\__, /     "
+	echo "  ________  ____  ___  /____/___ _/ /_____/____/_     "
+	echo " / ___/ _ \/ __ \/ _ \/ ___/ __  / __/ __ \/ ___/     "
+	echo "/ /__/  __/ / / /  __/ /  / /_/ / /_/ /_/ / /         "
+	echo "\___/\___/_/ /_/\___/_/   \__,_/\__/\____/_/          "
+	echo ""
+	echo "For Epix-Incorporated. Licensed under MIT"
+	echo ""
+fi
+
+git --no-pager log v$tagVer..HEAD --max-count=1024 --show-pulls --format="https://github.com/Epix-Incorporated/Adonis/commit/%H (Git/%an) %s" | grep "\\(#[0-9]+\\)$" -E -v | grep "(Update|Add|Change|Bump|Publish|v\-?[0-9][0-9A-Fa-fXx\.]*[ab]?|Changelog)\s*(rojo|Adonis|model|to)?\s*(Version|Changelog|update|v\-?[0-9])" -E -i -v > Adonis_TEMP_ReleaseNotes.txt
+if [ "$fromStdin" = "1" ]; then
+	cat > Adonis_TEMP_ReleaseNotes.txt
+elif [ "$doEcho" = "1" ]; then
+	read -p ""
+else
+	read -p "Please write the release notes (but not pulls) to Adonis_TEMP_ReleaseNotes.txt by examining the commits. DO NOT LEAVE AN UNFORMATTED CHANGELOG! When done, press Enter to continue..."
+fi
+if [ "$(tail -c1 Adonis_TEMP_ReleaseNotes.txt)" != "" ]; then
+	echo "Fatal error. File missing a newline!"
+    exit 1
+fi
+
+echo "	\"\";" > SigmaSigmaBoy.tmp
+printf "\t\"[%sv%s" "$verPrefix" "$newVer" >> SigmaSigmaBoy.tmp
+date +" %Y-%m-%d %T %Z" -u | tr -d "\n" >> SigmaSigmaBoy.tmp
+printf "] @" >> SigmaSigmaBoy.tmp
+git config user.name | sed -e 's/([Nn]ichole|([Dd]imenp?s[yi]onal|[Pp]bst?)[Ff]usion|[Dd]imenp?s[yi]onal)/Dimenpsyonal/g' | tr -d "\n" >> SigmaSigmaBoy.tmp
+echo "\";" >> SigmaSigmaBoy.tmp
+cat Adonis_TEMP_ReleaseNotes.txt | dos2unix -r -q --to-stdout | tr -d "\r" | sed 's/\\/\\\\/g; s/\"/\\\"/g' | sed -E 's/^.*$/\t\"\0\",/g' >> SigmaSigmaBoy.tmp
```

Comment:
Consider using `sed 's|\|\\|g; ...` to avoid having to escape the backslashes in situations like this.